drivers/hmis/spec/requests/hmis/hmis_activity_log_spec.rb

```diff
@@ -0,0 +1,145 @@
+    let(:project_query) do
+      <<~GRAPHQL
+        query #{operation_name}($id: ID!) {
+          project(id: $id) {
+            id
+            projectName
+            projectType
+            organization {
+              id
+            }
+          }
+        }
+      GRAPHQL
+    end
+
+    let!(:p2) { create :hmis_hud_project, data_source: ds1, organization: o1, user: u1 }
+    before(:each) do
+      create_access_control(hmis_user, p1)
+      create_access_control(hmis_user, p2)
+      hmis_login(user)
+    end
+
+    it 'should log a multiplexed query' do
+      queries = [
+        {
+          query: project_query,
+          variables: { 'id' => p1.id.to_s },
+          operation_name: operation_name,
+        },
+        {
+          query: project_query,
+          variables: { 'id' => p2.id.to_s },
+          operation_name: operation_name,
+        },
+      ]
+
+      expect do
+        response, _result = post_graphql_multi(queries: queries, headers: headers)
+        expect(response.status).to eq 200
+      end.to change(Hmis::ActivityLog, :count).by(2)
+
+      queries.zip(Hmis::ActivityLog.order(:id).last(2)).each do |query, log|
+        check_log(
+          log,
+          user: user,
+          variables: query[:variables],
+          operation_name: query[:operation_name],
+          headers: headers,
+        )
+        check_project_fields(log, project_id: query.dig(:variables, 'id'))
+      end
+    end
+
+    it 'should log a single query' do
+      variables = { 'id' => p1.id.to_s }
+      expect do
+        response, _result = post_graphql_single(variables: variables, headers: headers, operation_name: operation_name, query: project_query)
+        expect(response.status).to eq 200
+      end.to change(Hmis::ActivityLog, :count).by(1)
+
+      log = Hmis::ActivityLog.order(:id).last
+      check_log(log, user: user, variables: variables, operation_name: operation_name, headers: headers)
+      check_project_fields(log, project_id: p1.id)
+    end
+  end
+
+  describe 'access resolver' do
+    let(:operation_name) { 'getTestProject' }
+    before(:each) do
+      hmis_login(user)
+    end
+    let(:access_query) do
+      <<~GRAPHQL
+        query #{operation_name}{
+          access {
+            id
+            canAdministerHmis
+            canEditClients
+            __typename
+          }
+        }
+      GRAPHQL
+    end
+
+    it 'should log queries' do
+      expect do
+        response, _result = post_graphql_single(headers: headers, operation_name: operation_name, query: access_query)
+        expect(response.status).to eq 200
+      end.to change(Hmis::ActivityLog, :count).by(1)
+
+      log = Hmis::ActivityLog.order(:id).last
+      check_log(log, user: user, operation_name: operation_name, headers: headers)
+    end
+  end
+
+  def check_log(log, user:, variables: nil, operation_name:, headers:)
+    expect(log.user_id).to eq(user.id)
+    expect(log.ip_address).to eq('127.0.0.1')
```

Comment:
@ttoomey, somewhere you had a question about ip_address.  I'm not finding it, but for the warehouse we use `request.remote_ip` instead of `ip_address` not sure if that helps or not.  Other notes that might be useful are here:
`app/controllers/concerns/activity_logger.rb`

Reply:
Oh yeah, it was in the graphql_controller. We use `request.remote_ip` there - I'm just not sure how that will behave in the HMIS since the request comes through a proxy.